Bugzilla.pm

```diff
@@ -588,7 +588,7 @@ sub log_user_request {
   my $cgi         = $class->cgi;
   my $user_id     = $class->user->id;
   my $request_url = $cgi->request_uri // '';
-  my $method      = $cgi->request_method;
+  my $method      = $cgi->request_method // 'GET';
```

Comment:
Is `$cgi` from a 3rd party? If not, perhaps we should default `request_method` to `'GET'` earlier in the request lifecycle.

Reply:
Yeah it comes from upstream CGI Perl module
https://metacpan.org/release/LEEJO/CGI-4.53/source/lib/CGI.pm#L2902

So in our case some requests, the environment $ENV{'REQUEST_METHOD'} is not getting set properly and is undefined.

Reply:
> So in our case some requests, the environment $ENV{'REQUEST_METHOD'} is not getting set properly and is undefined.

This might be worth investigating further to see why it's not getting set properly.